UI/Web/src/app/book-reader/_components/book-reader/book-reader.component.ts

```diff
@@ -1545,7 +1554,44 @@ export class BookReaderComponent implements OnInit, AfterViewInit, OnDestroy {
     return side === 'right' ? 'highlight-2' : 'highlight';
   }
 
+  private mapMousePositionToPointerCursor(event: MouseEvent): boolean {
+    if (!this.isLoading && this.clickToPaginate) {
+      if (event.clientX <= window.innerWidth * 0.2 || event.clientX >= window.innerWidth * 0.8) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    return false;
+  }
+
+  private changeCursor(pointer: boolean) {
+    this.pointer = pointer;
+    this.cdRef.markForCheck();
+  }
 
+  handleReaderClick(event: MouseEvent) {
```

Comment:
At first glance, you can rewrite this as:

```
handleReaderClick(event: MouseEvent) {
    if (!this.clickToPaginate) { return; }

    const probablyHighlighting = window.getSelection()?.toString() != '';

    console.log('probablyHighlighting: ', probablyHighlighting);
    if (event.clientX <= window.innerWidth * 0.2) {
      if (!probablyHighlighting) {
        this.movePage(this.readingDirection === ReadingDirection.LeftToRight ? PAGING_DIRECTION.BACKWARDS : PAGING_DIRECTION.FORWARD);
        return;
      }
    } else if (event.clientX >= window.innerWidth * 0.8) {
      if (!probablyHighlighting) {
        this.movePage(this.readingDirection === ReadingDirection.LeftToRight ? PAGING_DIRECTION.FORWARD : PAGING_DIRECTION.BACKWARDS);
        return;
      }
    }
    
    this.toggleMenu(event);
  }
  ```
  
  This employs Inversion of Control to avoid nesting on this.clickToPaginate. We can avoid likely buggy math by just getting the selection from the browser, that will tell us if we are actually selecting something. We can use shortcircuiting to avoid if/else nesting to keep the logic clean.

Reply:
Rewrote it to be even more clean:

```
handleReaderClick(event: MouseEvent) {
    if (!this.clickToPaginate) { return; }

    const isHighlighting = window.getSelection()?.toString() != '';
    if (isHighlighting) {
      this.toggleMenu(event);
      return;
    }

    if (event.clientX <= window.innerWidth * 0.2) {
      this.movePage(this.readingDirection === ReadingDirection.LeftToRight ? PAGING_DIRECTION.BACKWARDS : PAGING_DIRECTION.FORWARD);
    } else if (event.clientX >= window.innerWidth * 0.8) {
      this.movePage(this.readingDirection === ReadingDirection.LeftToRight ? PAGING_DIRECTION.FORWARD : PAGING_DIRECTION.BACKWARDS);
    }
  }
  ```